provider/rotation_resource.go

```diff
@@ -0,0 +1,483 @@
+			rotation.Strategy.HandoffTime = ""
+			rotation.Strategy.HandoffDay = ""
+		} else {
+			if rotation.Strategy.HandoffTime == "" {
+				return diag.Errorf("firehydrant_rotation.strategy.handoff_time is required when strategy type is '%s'", rotation.Strategy.Type)
+			}
+			if rotation.Strategy.Type == "weekly" && rotation.Strategy.HandoffDay == "" {
+				return diag.Errorf("firehydrant_rotation.strategy.handoff_day is required when strategy type is '%s'", rotation.Strategy.Type)
+			}
+
+			// Discard unused values to avoid ambiguity.
+			rotation.Strategy.ShiftDuration = ""
+			rotation.StartTime = ""
+		}
+	}
+
+	// Create the rotation
+	createdRotation, err := firehydrantAPIClient.Rotations().Create(ctx, teamID, scheduleID, rotation)
+	if err != nil {
+		return diag.Errorf("Error creating rotation %s: %v", teamID, err)
+	}
+
+	// Set the rotation's ID in state
+	d.SetId(createdRotation.ID)
+
+	return readResourceFireHydrantRotation(ctx, d, m)
+}
+
+func updateResourceFireHydrantRotation(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	firehydrantAPIClient := m.(firehydrant.Client)
+
+	id := d.Id()
+	teamID := d.Get("team_id").(string)
+	scheduleID := d.Get("schedule_id").(string)
+	tflog.Debug(ctx, fmt.Sprintf("Update rotation: %s", id), map[string]interface{}{
+		"id":          id,
+		"team_id":     teamID,
+		"schedule_id": scheduleID,
+	})
+
+	// Initialize updateRequest with basic fields
+	updateRequest := firehydrant.UpdateRotationRequest{
+		Name:        d.Get("name").(string),
+		Description: d.Get("description").(string),
+	}
+
+	if v, ok := d.GetOk("slack_user_group_id"); ok {
+		updateRequest.SlackUserGroupID = v.(string)
+	}
+	if v, ok := d.GetOk("enable_slack_channel_notifications"); ok {
+		updateRequest.EnableSlackChannelNotifications = v.(bool)
+	}
+	if v, ok := d.GetOk("prevent_shift_deletion"); ok {
+		updateRequest.PreventShiftDeletion = v.(bool)
+	}
+	if v, ok := d.GetOk("color"); ok && v.(string) != "" {
+		updateRequest.Color = v.(string)
+	}
+
+	// Check if effective_at exists in raw config rather than state
+	if raw := d.GetRawConfig().GetAttr("effective_at"); !raw.IsNull() {
+		effectiveAtStr := raw.AsString()
+		effectiveAt, err := time.Parse(time.RFC3339, effectiveAtStr)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		// Only set effective_at if it's in the future
+		if effectiveAt.After(time.Now()) {
+			updateRequest.EffectiveAt = effectiveAt.Format(time.RFC3339)
+			tflog.Debug(ctx, "Rotation update will take effect at: "+updateRequest.EffectiveAt, map[string]interface{}{
+				"effective_at": updateRequest.EffectiveAt,
+			})
+		} else {
+			tflog.Debug(ctx, "Provided effective_at is in the past, update will take effect immediately", map[string]interface{}{
+				"effective_at": effectiveAtStr,
+				"now":          time.Now().Format(time.RFC3339),
+			})
+		}
+	}
+
+	// Get member IDs
+	inputMemberIDs := d.Get("members").([]interface{})
```

Comment:
Same concern as above :-)

Reply:
See answer above, but thanks for pointing this out in both places in case the change was needed.